src/coreclr/vm/ceemain.cpp

```diff
@@ -947,6 +950,21 @@ void EEStartupHelper()
                                                 g_MiniMetaDataBuffMaxSize, MEM_COMMIT, PAGE_READWRITE);
 #endif // FEATURE_MINIMETADATA_IN_TRIAGEDUMPS
 
+#ifdef TARGET_UNIX
+        // Preload the libSystem.Native.so/dylib to detect possible problems with loading it early
```

Comment:
Is this going to work for singlefilehost?

Reply:
Ah, singlefilehost should have been excluded here